spec/models/document_type_selection/option_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "json"
+
+RSpec.describe DocumentTypeSelection::Option do
+  describe ".managed_elsewhere?" do
+    it "returns true if the option is managed_elsewhere" do
+      option = {
+        "id" => "foo",
+        "type" => "managed_elsewhere",
+      }
+
+      expect(described_class.new(option).managed_elsewhere?).to be true
+    end
+  end
+
+  describe ".managed_elsewhere_url" do
+    let(:whitehall_host) { Plek.new.external_url_for("whitehall-admin") }
+
+    it "returns the path if a hostname is not provided" do
+      option = {
+        "id" => "foo",
+        "type" => "managed_elsewhere",
+        "path" => "/bar",
+      }
+
+      expect(described_class.new(option).managed_elsewhere_url).to eq("/bar")
+    end
+
+    it "returns the full url if the hostname is provided" do
+      option = {
+        "id" => "foo",
+        "type" => "managed_elsewhere",
+        "hostname" => "whitehall-admin",
+        "path" => "/bar",
+      }
+
+      expect(described_class.new(option).managed_elsewhere_url).to eq("#{whitehall_host}/bar")
```

Comment:
I was wondering where whitehall_host came from as I missed the let. As the variable is only used in this test it might be a simpler read being within the test.

You might also find that cutting plek out and writing the exact URL out gives the easiest read. Eg:
```
expect(described_class.new(option).managed_elsewhere_url)
  .to eq("https://whitehall-admin.test.gov.uk/bar")
```

Reply:
I like Plek because I don't think I should have to find out or care what the url is. 
I'll move assignment of `whitehall_host` into the test though.